src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,390 @@
+      min-height: 6rem;
+    }
+
+    .childContainer {
+      position: relative;
+      margin-bottom: 0.5rem;
+    }
+
+    .switchButton {
+      display: flex;
+      flex-direction: column;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      right: 0;
+      padding: 0.3rem;
+      margin: 0;
+      width: 2rem;
+      background: transparent;
+      border: none;
+    }
+
+    .smallText {
+      font-size: 0.875rem;
+      font-weight: 400;
+    }
+
+    .sectionTitle {
+      margin: 1rem 0 0.5rem 0;
+      color: black;
+    }
+
+    hr {
+      border-color: var(--grey-050);
+      margin: 1rem 0;
+    }
+
+    .not-displayed {
+      display: none;
+    }
+  </style>
+  <div class="container">
+    <h1>Authorize private login</h1>
+    <p class="sectionTitle">Application URL</p>
+    <div class="highlightBox smallText">${hostName}</div>
+    <p class="sectionTitle">Identity Anchor</p>
+    <div>
+      <div id="newUserSection" class="modeContainer">
+        <div class="childContainer">
+          <input
+            class="anchorText"
+            type="text"
+            id="userNumberInput"
+            placeholder="Enter anchor"
+          />
+          ${userNumber !== undefined
+            ? html` <button id="existingAnchorButton" class="switchButton">
+                ${undoIcon}
+              </button>`
+            : ""}
+        </div>
+        <div id="invalidAnchorMessage" class="error-message-hidden smallText">
+          The Identity Anchor is not valid. Please try again.
+        </div>
+      </div>
+      <div id="existingUserSection" class="modeContainer">
+        <div class="childContainer">
+          <div id="identityAnchor" class="highlightBox anchorText">
+            ${userNumber}
+          </div>
+          <button id="editAnchorButton" class="switchButton">
+            ${editIcon}
+          </button>
+        </div>
+      </div>
+    </div>
+    <button type="button" id="authorizeButton" class="primary">
+      Authorize
+    </button>
+    <div class="spacer"></div>
+    <div id="registerSection">
+      <div style="text-align: center">or</div>
+      <div class="spacer"></div>
+      <button type="button" id="registerButton">
+        Create New Identity Anchor
+      </button>
+      <div class="spacer-small"></div>
+    </div>
+    <div class="textLink">
+      <hr />
+      <div>
+        <button id="recoverButton" class="linkStyle">Lost access?</button>
+      </div>
+      <div>
+        <button type="button" class="linkStyle" id="manageButton">
+          Manage your Identity Anchor
+        </button>
+      </div>
+      <hr />
+    </div>
+    ${navbar}
+  </div>
+  ${footer}`;
+
+export class AuthSuccess {
+  constructor(
+    public userNumber: bigint,
+    public connection: IIConnection,
+    public sendDelegationMessage: () => void
+  ) {}
+}
+
+/**
+ * Shows the authorize application view and returns information about the authenticated user and a callback, which sends
```

Comment:
~~What's the "authorize application view"? I'd assume it's the first page that asks if you authorize the dapp to authenticate, but I don't see any of this happening in the code, so I'm guessing you mean something else?~~

---

Ah, I see. We first wait for the initial request, and _then_ load up the view.

Reply:
Yes, exactly.